client/home/lib/virtualmachines/components/machineslist/machinedetails.coffee

```diff
@@ -225,8 +225,15 @@ generateSpecs = (machine) ->
     't2.nano'   : '512MB RAM'
     't2.micro'  : '1GB RAM'
     't2.medium' : '4GB RAM'
+    'm4.large'  : '8GB RAM'
+    'm4.xlarge' : '16GB RAM'
+    'm4.2xlarge': '32GB RAM'
   }[type] ? '1GB RAM'
-  specs.push cpu = '1x CPU'
+  specs.push cpu = {
+    'm4.large'   : '2x CPU'
+    'm4.xlarge'  : '4x CPU'
+    'm4.2xlarge' : '8X CPU'
+  }[type] ? '1x CPU'
```

Comment:
not currently and this existing list also needed to be provider specific, these are only valid for `aws` provider only. I suggest to create a provider specific list somewhere else and not keep all these information in the view.